documentation/modules/oauth/con-oauth-authentication-flow.adoc

```diff
@@ -1,12 +1,91 @@
 // Module included in the following assemblies:
 //
-// assembly-oauth.adoc
+// assembly-oauth-authentication.adoc
 
 [id='con-oauth-authentication-flow-{context}']
-= OAuth 2.0 authentication mechanism
+= OAuth 2.0 authentication mechanisms
 
-The Kafka _SASL OAUTHBEARER_ mechanism is used to establish authenticated sessions with a Kafka broker.
+Strimzi supports the OAUTHBEARER and PLAIN mechanisms for OAuth 2.0 authentication. 
+Both mechanisms allow Kafka clients to establish authenticated sessions with Kafka brokers. 
+The authentication flow between clients, the authorization server, and Kafka brokers is different for each mechanism.
 
-A Kafka client initiates a session with the Kafka broker using the _SASL OAUTHBEARER_ mechanism for credentials exchange, where credentials take the form of an access token.
+Configure clients to use OAUTHBEARER whenever possible. 
+This mechanism provides a higher level of security because client credentials are _never_ shared with Kafka brokers. 
+Consider using PLAIN only with Kafka clients that do not support OAUTHBEARER. 
 
-Kafka brokers and clients need to be configured to use OAuth 2.0.
+If necessary, OAUTHBEARER and PLAIN can be enabled together, on the same `oauth` listener.
+
+.OAUTHBEARER overview
+
+OAUTHBEARER is the default authentication mechanism included as part of Kafka.
```

Comment:
If I didn't know better, I would understand this sentence as if OAUTHBEARER is the mechanism everyone uses with Kafka because all clients and server use it by default. That is of course not the case. OAUTHBEARER is supported, but has to be explicitly configured, and if you don't want to write your own callback handlers, you don't have many other options (no other option that I know of in fact) but to use the callbacks provided by Strimzi.

Reply:
I will update the explanation, thank you.